### DISCUSSION TITLE
Is it possible to set app state in automation action?

### DISCUSSION THREAD

**Discussion by mroziken** [Budibase/budibase#5061] [Help]
I would like to display a notification icon when new notification is registered. I thought I can use automation action to set app state as app state can be used in condition for showing/hiding items on the screen. I tried add JS Scripting with the following code. 

`$("state.newAlert") = True;`

but it does not seems to be set. Any idea how can I set app state? Or any other idea how can I control showing/hiding items on the screen bases on received trigger?

**Comment by mjashanks** [comment]
So - I think you are referring to the `Update State` action, when clicking a button. In which case it will look something like:

![image](https://user-images.githubusercontent.com/3524181/159882329-e3deee00-eca2-4236-8e74-aef7cad9a614.png)

Then the javascript will be:

![image](https://user-images.githubusercontent.com/3524181/159882398-e717c988-da6f-49a3-b31e-d8aa990b0209.png)

You also mention "Automation" - note that automations are server-side only - you cannot use automations to affect items on the screen.
